docs/wip-plugins.md

```diff
@@ -370,6 +388,121 @@ Returns `true` if the channel can be moderated by the current user.
 
 Returns `true` if the current user is a VIP in the channel.
 
+#### `HTTPMethod` enum
+
+This table describes HTTP methods available to Lua Plugins. The values behind
+the names may change, do not count on them. It has the following keys:
+
+- `Get`
+- `Post`
+- `Put`
+- `Delete`
+- `Patch`
+
+#### `HTTPResult`
+
+An `HTTPResult` is a table that contains the following keys:
+
+```lua
+{
+    data = "This is the data received from the server as a string",
```

Comment:
How do we see plugins use the returned data when it's only returned as a string?

Reply:
This isn't a unicode text string, it's handled more like a binary buffer, we convert a QByteArray into an std::string and give the plugin that. They can use it however they like.

Reply:
I was thinking more in terms of the usability of someone wanting to use a JSON API, which should be pretty common. How would that look like?

Reply:
Making a json parser api shouldn't be incredibly hard, it's mostly just:
 1. take string from lua
 2. parse with favorite json parser
 3. push result as a table into lua

For serialization just go other way around. As long as we don't have recursive structures, step 3 will be easy.